src/mono/mono/mini/aot-compiler.c

```diff
@@ -12368,7 +12414,7 @@ collect_methods (MonoAotCompile *acfg)
 		}
 		*/
 
-		if (method->is_generic || mono_class_is_gtd (method->klass)) {
+		if (should_emit_extra_method_for_generics (method, TRUE)) {
 			/* Compile the ref shared version instead */
```

Comment:
The aot runtime assumes that the index of 'normal' methods is the same as their token, which means that all normal methods need to be added.  They can fail compilation later.

Reply:
I am not sure if I quite understand this:
1. Are ref shared methods considered as normal?
2. If they are, how could we make a workaround on this assumption?
3. And finally, wouldn't this change already break some CI pipelines?

Reply:
I guess we add the method below using add_method_with_index () so it will work.

Reply:
Can this be then marked as resolved?

Reply:
So what happens here is we add the original generic method, and the JIT fails to compile it, since you can't compile uninflated generic methods. So the ids/indexes will be correct, since we do add these methods, but this looks a bit ugly. You can see this in action by running with 
--aot=print-skipped